### PR TITLE
Stage to master

### PR DESCRIPTION
**Pull request by jerryblakley**
Add CURLOPT_ERRORBUFFER to capture better error information. However, this may not apply to those '400' errors we were seeing on GET requests, since those are not 'curl errors'.
Also some additions to fusedav tests, as well as expansion of 'server' tests, which are essentially 'externalized' tests to replace python server unit tests.

### CODE REVIEW

**Comment by jerryblakley** [comment]
@kf6nux You may have reviewed a few weeks back, but if you have time, can you give it another gander? Thanks. I guess I still have some CircleCI stuff to do as well.

File: tests/server-tests/api_test.go [hunk 1]
```diff
@@ -101,20 +96,80 @@ func testMethod(t *testing.T, testInput TestInput) error {
...
+	testInput.method = "GET"
+
+	// for _, method := range []string{"HEAL", "TEST", "LOCK", "PROPPATCH", "OPTIONS"} {
+	// Temporarily remove HEAL, it takes a long time
+	for _, entry := range paths {
+		testInput.path = entry.path
+		testInput.expectedStatusCode = entry.statusCode
+		err := testMethod(t, testInput)
+		if err != nil {
+			// handle error
```
Comment (kf6nux) [inline tests/server-tests/api_test.go]: I'm not sure if ignoring an error here is going to cause pain in debugging later.

File: tests/server-tests/api_test.go [hunk 2]
```diff
@@ -203,16 +257,16 @@ func TestBasicFileOps(t *testing.T) {
...
 	}
 
+	// Clear the headers
+	testInput.headers = nil
+
 	testInput.method = "GET"
 	testInput.expectedStatusCode = 200
 	err = testMethod(t, testInput)
 	if err != nil {
 		// handle error
```
Comment (kf6nux) [inline tests/server-tests/api_test.go]: I'm not sure if ignoring an error here is going to cause pain in debugging later.

File: src/filecache.c [hunk 3]
```diff
@@ -531,7 +531,7 @@ static void get_fresh_fd(filecache_t *cache,
     // but on the close (dav_flush/release), the PUT fails and the file never makes it to the server.
     // On opening again, the server will deliver this unexpected 404. Changes for forensic-haven
     // should prevent these errors in the future (2013-08-29)
-    if (inject_error(filecache_error_fresh404)) response_code = 404;
+    if (inject_error(filecache_error_fresh404)) response_code = 400;
```
[the remark this answers is not in the document]
Comment (jerryblakley) [reply src/filecache.c]: Made the changes.

File: tests/server-tests/api_test.go [hunk 4]
```diff
@@ -82,7 +77,7 @@ func testMethod(t *testing.T, testInput TestInput) error {
 
 	for _, pair := range testInput.headers {
 		req.Header.Add(pair.key, pair.value)
-		fmt.Printf("testMethod: method: %s; path: %s; header: %v\n", testInput.method, testInput.path, req.Header)
+		fmt.Printf("testMethod: Headers: method: %s; path: %s; header: %v\n", testInput.method, testInput.path, req.Header)
```
[the remark this answers is not in the document]
Comment (jerryblakley) [reply tests/server-tests/api_test.go]: testMethod is the name of the function. It's a pattern I use in the fusedav code to list the function name in the log message. Not sure the value here. Leaving it in. As for 'Headers:', this is just an information message, and Headers is just further definition of what the message is conveying.

File: tests/server-tests/api_test.go [hunk 5]
```diff
@@ -101,20 +96,80 @@ func testMethod(t *testing.T, testInput TestInput) error {
...
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			// handle error
+			t.Errorf("testMethod: Error on ioutil.ReadAll: %v", err)
+		}
+
+		if testInput.expectedStatusCode != 404 &&
+			len(testInput.content) > 0 &&
+			string(body) != testInput.content {
+			t.Errorf("testMethod: Error, expected content %v, got %v", testInput.content, resp.Body)
```
[the remark this answers is not in the document]
Comment (jerryblakley) [reply tests/server-tests/api_test.go]: Yup

File: tests/server-tests/api_test.go [hunk 6]
```diff
@@ -101,20 +96,80 @@ func testMethod(t *testing.T, testInput TestInput) error {
...
+
+	var testInput TestInput
+	var testOutput TestOutput
+	var paths []PathResult
+
+	fmt.Println("TestPaths")
+
+	testInput.client = getClient()
+
+	paths = make([]PathResult, 7)
```
[the remark this answers is not in the document]
Comment (jerryblakley) [reply tests/server-tests/api_test.go]: paths[0] is not a constant, and all other values build on it. Not sure how to put together the literal.

**Pull request merged**